tests/helpers/test_template.py

```diff
@@ -41,13 +40,64 @@ def tearDown(self):
         """Stop down stuff we started."""
         self.hass.stop()
 
+    def generate_filter(self, template_str, variables=None):
+        """Create filter from template."""
+        tmp = template.Template(template_str, self.hass)
+        (_, filt) = tmp.async_render_with_collect(variables)
```

Comment:
This should use the sync version really.

Reply:
We still need to keep sync versions of the core though, and templates could well be used by some sync component. Maybe clean it up at the end of #22587, including changing the test cases?

The downside of changing over the test cases is that then the sync path doesn't get tested.

Reply:
OK what I propose is leave the test cases using the sync version for the moment, but once I've got rid of all the old code (`extract_entities`) then change the test cases over to async test cases as well. Then we can remove the sync path cleanly.